### ISSUE TITLE
Inverse match returns no result when matched fields are nullish

### ISSUE THREAD

**Issue by asnaeb** [krisk/Fuse#749]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Description of the bug

When using reverse matching operator `!` along with nested search, items with matched fields that have nullish values (`undefined`, `null` or `""`) are completely discarded from the results. I think that a reverse match operator should match nullish values too. `""`, `null` and `undefined` **should** match `!somestring`! Is there a workaround for this that I am missing?

### The Fuse.js version where this bug is happening.

6.6.2

### Is this a regression?

- [ ] This is a regression bug

### Which version did this behavior use to work in?

None

### Steps To Reproduce
We try to match every item whose `lastName` field doesn't include the string `doe`
```js
const items = [
  {
    firstName: "John",
    lastName: null
  },
  {
    firstName: "Jack",
    lastName: undefined
  },
  {
    firstName: "Jim",
    lastName: ""
  }
];

const fuse = new Fuse(items, {
  keys: [ "firstName", "lastName"],
  useExtendedSearch: true
});

const result = fuse.search({lastName: "!doe"});

result.length === 0; // true;
```
Result is empty, but none of the three items actually have a `lastName` field which includes `doe`.
### Expected behavior
```js
const items = [
  {
    firstName: "John",
    lastName: null
  },
  {
    firstName: "Jack",
    lastName: undefined
  },
  {
    firstName: "Jim",
    lastName: ""
  }
];

const fuse = new Fuse(items, {
  keys: [ "firstName", "lastName"],
  useExtendedSearch: true
});

const result = fuse.search({lastName: "!doe"});

result.length === 3;
```

**Comment by asnaeb** [comment]
For anyone landing here, This is the workaround I am using right now
```js
const options = {
  getFn(...args) {
    return Fuse.config.getFn(...args) || [];
  }
};

new Fuse(myCollection, options);
```

This will prevent nullish values to be excluded from the results

**Comment by asnaeb** [comment]
For anyone landing here with the same problem, I am currently using the following workaround
```js
const fuse = new Fuse(items, {
  ...yourOptions,
  useExtendedSearch: true,
  getFn(...args) {
    return Fuse.config.getFn(...args) || "\uFFFF";
  }
});
```
This will return the unicode non-character `FFFF`as a fallback for nullish values, preventing them to be excluded. This character is very unlikely to be typed inside a search string so it is working fine for me at the moment but still...

**Comment by fhollste** [comment]
I am facing this issue as well but the suggested workaround does not work in my case. I believe that this is due to searching in a nested property where the parent is undefined. I am using a fork so I could fix it locally, can you @asnaeb give any hints on what causes the issue? Thanks!

**Comment by krisk** [comment]
Fixed in f823353e. When a field value is null, undefined, or missing, inverse matchers (`!`, `!^`, `!..$`) now correctly treat it as a match — a missing value does not contain the search term.

**Comment by krisk** [comment]
After further consideration, the fix was reverted. When a field is null/undefined, the data doesn't exist — the field shouldn't participate in the search at all, including inverse matches. Returning items with missing fields for `!doe` would flood results with irrelevant items that happen to have incomplete data.

If you need to filter for items where a field is missing or doesn't contain a value, we'd recommend handling that in application code before or after the search.
